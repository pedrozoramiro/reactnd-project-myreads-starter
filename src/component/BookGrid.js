import React, { Component } from "react";

import PropTypes from "prop-types";

class BookGrid extends Component {
  render() {
    const { title, books, actions } = this.props;
    
    return (
      <div className="bookshelf">
        <h2 className="bookshelf-title">{title}</h2>
        <div className="bookshelf-books">
          <ol className="books-grid">
            {books.map(function(book) {
              return (
                <li key={book.id}>
                  <div className="book">
                    <div className="book-top">
                      <div
                        className="book-cover"
                        style={{
                          width: 128,
                          height: 193,
                          backgroundImage: `url(${book.imageLinks
                            .smallThumbnail})`
                        }}
                      />
                      <div className="book-shelf-changer">
                        <select>
                          <option value="none" disabled>
                            Move to...
                          </option>
                          {actions.map(function(act) {
                            return (
                              <option key={act.key} value={act.key} >{act.title}</option>
                            );
                          })}
                        </select>
                      </div>
                    </div>
                    <div className="book-title">{book.title}</div>
                    <div className="book-authors">{book.authors}</div>
                  </div>
                </li>
              );
            })}
          </ol>
        </div>
      </div>
    );
  }
}

BookGrid.propTypes = {
  title: PropTypes.string.isRequired
};

export default BookGrid;
