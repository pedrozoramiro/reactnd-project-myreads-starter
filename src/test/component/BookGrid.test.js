import React from "react";
import { shallow, mount } from "enzyme";

import BookGrid from "../../component/BookGrid";


const mock = {
  books:[
    {
      title: "Book",
      shelf: "read",
      imageLinks: { smallThumbnail: "" },
      authors: ["Ramiro Pedrozo"]
    }
  ] ,
  shelves: [
    { name: "currentlyReading", title: "Currently Reading" },
    { name: "wantToRead", title: "Want to Read" },
    { name: "read", title: "Read" }
  ],
  handleChangeShelf: jest.fn()
};

it("renders without crashing", () => {
  const bookWrapper = (
    <BookGrid
      books={mock.books}
      shelves={mock.shelves}
      handleChangeShelf={mock.handleChangeShelf}
    />
  );
  expect(shallow(bookWrapper)).toMatchSnapshot();
});
