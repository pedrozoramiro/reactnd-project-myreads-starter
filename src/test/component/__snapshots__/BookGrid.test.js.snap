// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<div
  className="app"
>
  <div
    className="list-books"
  >
    <div
      className="list-books-title"
    >
      <h1>
        MyReads
      </h1>
    </div>
    <div
      className="list-books-content"
    >
      <div>
        <BookShelf
          books={
            Array [
              Object {
                "authors": Array [
                  "Ramiro Pedrozo",
                ],
                "imageLinks": Object {
                  "smallThumbnail": "",
                },
                "shelf": "read",
                "title": "Book",
              },
            ]
          }
          handleChangeShelf={[Function]}
          key="currentlyReading"
          shelf={
            Object {
              "name": "currentlyReading",
              "title": "Currently Reading",
            }
          }
          shelves={
            Array [
              Object {
                "name": "currentlyReading",
                "title": "Currently Reading",
              },
              Object {
                "name": "wantToRead",
                "title": "Want to Read",
              },
              Object {
                "name": "read",
                "title": "Read",
              },
            ]
          }
        />
        <BookShelf
          books={
            Array [
              Object {
                "authors": Array [
                  "Ramiro Pedrozo",
                ],
                "imageLinks": Object {
                  "smallThumbnail": "",
                },
                "shelf": "read",
                "title": "Book",
              },
            ]
          }
          handleChangeShelf={[Function]}
          key="wantToRead"
          shelf={
            Object {
              "name": "wantToRead",
              "title": "Want to Read",
            }
          }
          shelves={
            Array [
              Object {
                "name": "currentlyReading",
                "title": "Currently Reading",
              },
              Object {
                "name": "wantToRead",
                "title": "Want to Read",
              },
              Object {
                "name": "read",
                "title": "Read",
              },
            ]
          }
        />
        <BookShelf
          books={
            Array [
              Object {
                "authors": Array [
                  "Ramiro Pedrozo",
                ],
                "imageLinks": Object {
                  "smallThumbnail": "",
                },
                "shelf": "read",
                "title": "Book",
              },
            ]
          }
          handleChangeShelf={[Function]}
          key="read"
          shelf={
            Object {
              "name": "read",
              "title": "Read",
            }
          }
          shelves={
            Array [
              Object {
                "name": "currentlyReading",
                "title": "Currently Reading",
              },
              Object {
                "name": "wantToRead",
                "title": "Want to Read",
              },
              Object {
                "name": "read",
                "title": "Read",
              },
            ]
          }
        />
      </div>
    </div>
    <div
      className="open-search"
    >
      <Link
        replace={false}
        to="/search"
      >
        Add a book
      </Link>
    </div>
  </div>
  }
</div>
`;
